code/parachain/frame/staking-rewards/src/test/mod.rs

```diff
@@ -271,6 +277,101 @@ fn stake_in_case_of_low_balance_should_not_work() {
 	});
 }
 
+#[test]
+fn stake_should_fail_if_amount_is_less_than_minimum() {
+	new_test_ext().execute_with(|| {
+		let staker = ALICE;
+		let amount = 9_000_u128;
+		let duration = ONE_HOUR;
+		let pool_id = PICA::ID;
+
+		process_and_progress_blocks::<StakingRewards, Test>(1);
+		assert_ok!(StakingRewards::create_reward_pool(Origin::root(), get_default_reward_pool()));
+
+		assert_noop!(
+			StakingRewards::stake(Origin::signed(staker), pool_id, amount, duration),
+			crate::Error::<Test>::StakedAmountTooLow
+		);
+	});
+}
+
+#[test]
+fn split_should_fail_if_any_amount_is_less_than_minimum() {
+	new_test_ext().execute_with(|| {
+		let staker = ALICE;
+
+		process_and_progress_blocks::<StakingRewards, Test>(1);
+
+		mint_assets([ALICE], [PICA::ID], PICA::units(10_000));
+
+		create_rewards_pool_and_assert::<Test, runtime::Event>(RewardRateBasedIncentive {
+			owner: ALICE,
+			asset_id: PICA::ID,
+			start_block: 2,
+			end_block: 5,
+			reward_configs: default_reward_config(),
+			lock: default_lock_config(),
+			share_asset_id: XPICA::ID,
+			financial_nft_asset_id: STAKING_FNFT_COLLECTION_ID,
+			minimum_staking_amount: MINIMUM_STAKING_AMOUNT,
+		});
+
+		process_and_progress_blocks::<StakingRewards, Test>(1);
+
+		let amount = 15_000;
+		let original_fnft_instance_id =
+			stake_and_assert::<Test, runtime::Event>(ALICE, PICA::ID, amount, ONE_MINUTE);
+
+		// Original stake is less than the minimum and new stake is greater.
+		let ratio = Permill::from_parts(333_334);
+
+		assert_eq!(ratio.mul_floor(amount), 5_000);
+		assert_eq!(ratio.left_from_one().mul_ceil(amount), 10_000);
+
+		assert_noop!(
+			StakingRewards::split(
+				Origin::signed(staker),
+				STAKING_FNFT_COLLECTION_ID,
+				original_fnft_instance_id,
+				ratio.try_into_validated().unwrap(),
+			),
+			crate::Error::<Test>::StakedAmountTooLowAfterSplit
+		);
+
+		// New stake is less than the minimum and original stake is greater.
+		let ratio = Permill::from_parts(666_667);
+
+		assert_eq!(ratio.mul_floor(amount), 10_000);
+		assert_eq!(ratio.left_from_one().mul_ceil(amount), 5_000);
+
+		assert_noop!(
+			StakingRewards::split(
+				Origin::signed(staker),
+				STAKING_FNFT_COLLECTION_ID,
+				original_fnft_instance_id,
+				ratio.try_into_validated().unwrap(),
+			),
+			crate::Error::<Test>::StakedAmountTooLowAfterSplit
+		);
+
+		// Both stakes are less than the minimum.
+		let ratio = Permill::from_parts(500_000);
+
+		assert_eq!(ratio.mul_floor(amount), 7_500);
+		assert_eq!(ratio.left_from_one().mul_ceil(amount), 7_500);
+
+		assert_noop!(
+			StakingRewards::split(
+				Origin::signed(staker),
+				STAKING_FNFT_COLLECTION_ID,
+				original_fnft_instance_id,
+				ratio.try_into_validated().unwrap(),
+			),
+			crate::Error::<Test>::StakedAmountTooLowAfterSplit
+		);
```

Comment:
NIT
These are three different test conditions, lets split them into their respective tests

Reply:
Seems like too repetitive to have 3 tests here, as they are basically the same but just making sure that under any non desirable scenario the operation will fail. Any better way of doing it without creating multiple tests? Maybe iteration over a small array?

Reply:
Prop testing with set values is an option but may be outside the scope of this PR.
In general, even if tests are repetitive in structure, it's good to have them split up if they are testing different cases. This will make debugging quicker if one but not all cases begin failing.
Up to you though, still a NIT comment.